docs/docs/documents/predefined-pages.mdx

```diff
@@ -0,0 +1,270 @@
+
+**Example:** The news page in the code at `https://example.com/news` should be rendered in the page tree at `https://example.com/company/news`.
+We need the following redirect and rewrite:
+
+-   A **rewrite** from `/company/news` to `/news` (to ensure that the page can be viewed at the expected location)
+-   A **redirect** from `/news` to `/company/news` (to avoid duplicate content)
+
+Perform the following changes in the middleware:
+
+1.  Copy the predefined pages [helper functions](https://github.com/vivid-planet/comet/tree/main/demo/site/src/middleware/predefinedPages.ts) from the Demo project
+
+    :::warning
+
+    Support for filtering by document type in the `paginatedPageTreeNodes` query has been added in version 7.4.0.
+    Make sure your project uses Comet 7.4.0 or later.
+
+    :::
+
+2.  Add the rewrite and redirect to the middleware:
+
+    ```diff
+    + import { getPredefinedPageRedirect, getPredefinedPageRewrite } from "./middleware/predefinedPages";
+
+    export async function middleware(request: NextRequest) {
+        /* ... */
+
+    +   const predefinedPageRedirect = await getPredefinedPageRedirect(scope, pathname);
+    +
+    +   if (predefinedPageRedirect) {
+    +       return NextResponse.redirect(new URL(predefinedPageRedirect, request.url), 307);
+    +   }
+    +
+    +   const predefinedPageRewrite = await getPredefinedPageRewrite(scope, pathname);
+    +
+    +   if (predefinedPageRewrite) {
+    +       return NextResponse.rewrite(new URL(predefinedPageRewrite, request.url));
+    +   }
+
+        return NextResponse.next();
+    }
+    ```
+
+#### Adding the predefined page document type to the `PageLink`
+
+Add the newly created document type to the `PageLink` component for links to work correctly:
+
+```diff title="src/layout/header/PageLink.tsx"
+function PageLink({ page, children, className: passedClassName, activeClassName, }: Props): JSX.Element | null {
+    /* ... */
+
+    if (page.documentType === "Link") {
+        /* ... */
+    } else if (page.documentType === "Page") {
+        /* ... */
++   } else if (page.documentType === "PredefinedPage") {
++       return (
++           <Link href={`/${page.scope.language}${page.path}`} className={className}>
++               {children}
++           </Link>
++       );
+    }
+}
+```
+
+Now, all the necessary changes should have been made to the project.
+Add a new page tree node with the document type "Predefined Page" to test whether it works correctly in the Site.
+
+## Adding a new predefined page type
+
+If you want to add a new type (e.g., products), the following changes must be made:
+
+1. API: Add the new type to the GraphQL enum:
+
+    ```diff title="src/documents/predefined-pages/entities/predefined-page.entity.ts"
+    export enum PredefinedPageType {
+        News = "News",
+    +   Products = "Products",
```

Comment:
in the introduction note you write "This guide uses news as an example of structured data", you should also use news here, I suppose

Reply:
This section is intended to show what's necessary to add another predefined page type, so I used products as a second example. Isn't this clear enough? I could switch products and news here though.

Reply:
I think it's okay. At least it was clear to me

Reply:
984e8193c46cf0862ed1fa57dff0504b919b4567